### PR TITLE
Add support for buildpack api 0.8

### LINKED ISSUE #128: Support Buildpacks API 0.8: switch from build/detect args to env variables
- When using buildpack API 0.8 or greater, the platform directory and build plan path are provided as environment variables to buildpack bin/detect, in addition to the deprecated positional arguments (https://github.com/buildpacks/lifecycle/pull/818)
- When using buildpack API 0.8 or greater, the platform directory, buildpack plan path, and buildpack layers directory are provided as environment variables to buildpack bin/build, in addition to the deprecated positional arguments (https://github.com/buildpacks/lifecycle/pull/818)

### PR DESCRIPTION
**Pull request by samj1912**
Resolves #128

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -191,11 +186,36 @@ func Build(builder Builder, options ...Option) {
...
+		}
+		buildpackPlanPath, ok = os.LookupEnv("CNB_BP_PLAN_PATH")
+		if !ok {
+			config.exitHandler.Error(fmt.Errorf("expected CNB_BP_PLAN_PATH to be set"))
+			return
+		}
+	}
 
-	ctx.Platform.Path = config.arguments[2]
+	logger.Debugf("Layers: %+v", ctx.Layers)
```
Comment (dmikusa-pivotal) [inline build.go:218]:
```suggestion
	logger.Debugf("Layers Directory: %+v", ctx.Layers)
```

**Review by dmikusa-pivotal — APPROVED** [review]
Looks good to me.

I'm not 100% sure it's necessary, but my only observation is that in the detect & build tests, we're not testing the case where both args & env variables are set.

I believe that the code is going to take the env variables if both are set. It's an invalid state, so it would seem reasonable if we errored if both are set. I don't know if the platform/lifecycle would allow this (I didn't test), but my line of thinking is that it might be possible for a user to set one of these new env variables on an older platform/lifecycle since the older platform/lifecycle is unaware of them altering the behavior. In this particular case, erroring out would be beneficial.

**Comment by samj1912** [comment]
> Looks good to me.
> 
> I'm not 100% sure it's necessary, but my only observation is that in the detect & build tests, we're not testing the case where both args & env variables are set.
> 
> I believe that the code is going to take the env variables if both are set. It's an invalid state, so it would seem reasonable if we errored if both are set. I don't know if the platform/lifecycle would allow this (I didn't test), but my line of thinking is that it might be possible for a user to set one of these new env variables on an older platform/lifecycle since the older platform/lifecycle is unaware of them altering the behavior. In this particular case, erroring out would be beneficial.

Even if the user sets this on an older version of the platform/lifecycle, we only source from the env var if the buildpack api >= 0.8.

File: build.go (hunk 1 above)
Comment (samj1912) [reply build.go:218]: Is it okay if we keep it to the previous statement since that is what libcnb was logging before?

Comment (dmikusa-pivotal) [reply build.go:218]: Oh, I thought that had changed in the diff. Yes, not a huge issue. Just thought that would be a little clearer.

**Pull request merged**